### ISSUE TITLE
Docu on building APIs

### ISSUE THREAD

**Issue by mpscholten** [digitallyinduced/ihp#506]

**Comment by pallavithakare28** [comment]
Hello @mpscholten, Is this issue still open? I would like to work on this issue.

**Comment by mpscholten** [comment]
Hey, yes this issue is still open 👍

**Comment by sycured** [comment]
Do you have an ETA?

**Comment by simg** [comment]
what will be overall architecture for the APIs? eg REST or GraphQL?  

your post implies it's just going to be the web form fields but converted to JSON? which is still useful but it may prove to be a wasted opportunity not to support a defacto standard like Swagger or GraphQL?

**Comment by mpscholten** [comment]
Higher-level API layers could be build on top of this. At digitally induced we so far only built "small" APIs with IHP, for these cases the JSON approach has worked well.

Swagger and GraphQL are definitely technologies we should offer as part of the IHP core in the future. But to add these we'd need to do this on a real-world project and then extract this into IHP.

**Comment by simg** [comment]
fair enough, I quite understand that it would be a non-trivial undertaking.

**Comment by G4BB3R** [comment]
I need JWT auth for a hobby API project. My currently case is a very common and simple. At login, it encodes a JWT with user id (and can contain other simple data in the future) and send the token as the JSON response { token: "..." }. Then the client captures it and store it, to send to the next protected requests. Then IHP could have a default middleware to decode and ensure it has the right permission in some requests, but the middleware could be replaced by a custom made. I will try to implement with the current framework, and post more details.
